### PR TITLE
Refined the ACL/Permission APIs according to review comments.

### PR DESCRIPTION
**Pull request by katmsft**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

File: sdk/storage/azure-storage-files-datalake/CHANGELOG.md [hunk 1]
```diff
@@ -2,9 +2,14 @@
...
 ## 12.0.0-beta.7 (Unreleased)
 
+### New Features
+
+- Added `Owner`, `Permissions`, and `Group` to `GetDataLakePathAccessControlResult`.
+
 ### Breaking Changes
 
 - Removed `GetDfsUri` in all clients since they are currently implementation details.
+- `DataLakePathClient` can no longer set permissions with `SetAccessControl`, instead, a new API `SetPermission` is created for such functionality. Renamed the original API to `SetAccessControlList` to be more precise.
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/CHANGELOG.md:12]: SetPermissions?

File: sdk/storage/azure-storage-files-datalake/src/datalake_path_client.cpp [hunk 2]
```diff
@@ -395,6 +412,18 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
       throw std::runtime_error("Got null value returned when getting access control.");
     }
     ret.Acls = std::move(acl.GetValue());
+    if (result->Owner.HasValue())
+    {
+      ret.Owner = result->Owner.GetValue();
+    }
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/src/datalake_path_client.cpp:418]: I don't understand, if this is not nullable in convenience layer, why is it nullable in protocol layer?

**Review by Jinming-Hu — CHANGES_REQUESTED** [review]

File: sdk/storage/azure-storage-files-datalake/src/datalake_path_client.cpp (hunk 2 above)
Comment (katmsft) [reply sdk/storage/azure-storage-files-datalake/src/datalake_path_client.cpp:418]: This convenience API is calling a protocol layer API shared by several convenience APIs. Please take a look at: https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/update

For some actions, owner is always in response if it exists.

**Review by Jinming-Hu — APPROVED** [review]

**Pull request merged**